src/system-fetch.js

```diff
@@ -1,5 +1,31 @@
   var fetchTextFromURL;
-  if (typeof XMLHttpRequest != 'undefined') {
+  if (typeof fetch != 'undefined') {
```

Comment:
Can we do this check after the `XMLHttpRequest` check to ensure it is only used when needed?